adminSiteServer/apiRoutes/mdims.ts

```diff
@@ -93,19 +95,23 @@ export async function handlePatchMultiDim(
     }
     const { published, slug } = req.body
     let action
-    if (published !== undefined && published !== multiDim.published) {
-        multiDim = await setMultiDimPublished(trx, multiDim, published)
-        action = published ? "publish" : "unpublish"
-    }
     if (slug !== undefined && slug !== multiDim.slug) {
-        await validateGrapherSlug(trx, slug)
+        await validateNewGrapherSlug(trx, slug)
         multiDim = await setMultiDimSlug(trx, multiDim, slug)
-        if (!action && multiDim.published) {
+        if (multiDim.published) {
             action = "publish"
         }
     }
+    // Note: Keep this change last, since we don't want to update the configs
```

Comment:
I don't know precisely when a DB rollback will happen, honestly.
Could it happen that we change the slug (and propagate that to R2), and then roll that back in the DB afterwards?

Reply:
It can still happen, it can roll back any time we throw an error before the helper knex function somewhere in our Express handler returns. But the later we do the R2 upload, the better.

We have this problem in many places, and it's fine. The proper way to do this would be with https://en.wikipedia.org/wiki/Change_data_capture, if we really needed the correctness, i.e. first commit the changes to the DB and then propagate the changes to external systems.